invenio_damap/services/services.py

```diff
@@ -33,68 +36,70 @@ def schema(self):
-
-    def add_record_to_dmp(self, identity, recid, dmp_id, data):
-        """Add the provided record to the DMP"""
-
-        person_id = self._get_user_id(identity=identity)
-        headers = self._create_headers(identity)
-
-        # this will also perform permission checks, ensuring the user may access the record.
-        record = current_rdm_records_service.read(identity, recid)
-        exported_record = InvenioDAMAPExport.export_as_madmp(record, **data)
-
-        r = requests.post(
-            url=self.config.damap_base_url
-            + "/api/invenio-damap/dmps/{}/{}".format(dmp_id, person_id),
-            headers=headers,
-            json=exported_record,
+    def _create_auth_jwt(self, identity, expires_in=600):
+        """Creates an authorization jwt token for DAMAP."""
+        person_data = self.config.damap_person_function(identity)
+
+        if person_data is None:
+            return
+
+        # more info about JWT claims: https://www.rfc-editor.org/rfc/rfc7519.html#section-4
+        return jwt.encode(
+            {
+                "sub": str(current_user.id),
+                "exp": int(time()) + expires_in,
+                "iat": int(time()),
+                "invenio-damap": {
+                    "identifiers": {
+                        **person_data,
+                    }
+                },
+            },
+            self.config.damap_shared_secret,
+            algorithm="HS256",
         )
 
-        r.raise_for_status()
-
-        return record
+    def _create_headers(self, identity, *args, **kwargs):
+        """Creates the auth header and additional ones, if defined."""
+        headers = self.config.damap_custom_header_function(identity=identity)
+        return headers
 
-    def search(self, identity, params):
+    def search(self, identity, params, jwt=None, **kwargs):
         """Perform search for DMPs."""
         self.require_permission(identity, "read")
 
         search_params = self._get_search_params(params)
-        person_id = self._get_user_id(identity=identity)
         headers = self._create_headers(identity)
+        headers.update(
+            {"X-Auth": (f"{jwt}" if jwt else f"{self._create_auth_jwt(identity)}")}
+        )
 
         r = requests.get(
-            url=self.config.damap_base_url
-            + "/api/invenio-damap/dmps/person/{}".format(person_id),
+            url=self.config.damap_base_url + "/api/madmps",
             headers=headers,
             params=search_params,
         )
-
-        r.raise_for_status()
+        # We don't want to trigger the exceptions on these codes.
+        # These indicate no valid identity between the 2 systems or no DMPs.
+        if r.status_code not in [401, 403, 404]:
+            r.raise_for_status()
 
         dmps = Pagination(
             query=None,
-            items=r.json(),
+            # If response is not 2xx, DAMAP won't return a JSON response.
+            items=r.json() if r.ok else [],
```

Comment:
Still not sure if it makes sense to not raise with another exception in these cases. The client would not know what really happened and can not act accordingly. @max-moser what do you think?